src/main/scala/org/graphframes/lib/ConnectedComponents.scala

```diff
@@ -336,8 +338,11 @@ object ConnectedComponents extends Logging {
 
         // remove previous checkpoint
         if (iteration > checkpointInterval) {
-          FileSystem.get(sc.hadoopConfiguration)
-            .delete(new Path(s"${checkpointDir.get}/${iteration - checkpointInterval}"), true)
+          val path = new Path(s"${checkpointDir.get}/${iteration - checkpointInterval}")
+          if (fileSystem == null) {
```

Comment:
From what I can see
- DStreamCheckpointData.scala does it this way but its history is hard to track down since it is moved in 2014
- this is what the call [leads to](https://github.com/apache/hadoop/blob/trunk/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileSystem.java#L3285) - basically it hits a lock multiple times mostly to cached the returned FileSystem

Since it is caching already let's move to not store it.